### ISSUE TITLE
Unable to connect to the remote server (Azure Ml compute instance)

### ISSUE THREAD

**Issue by kbokh** [microsoft/vscode-tools-for-ai#2244]
## Expected Behavior

Running a notebook on a Azure Ml compute instance using a local code editor

## Actual Behavior
 The message is shown: "Unable to connect to the remote server"

## Steps to Reproduce the Problem

  1. Open a notebook
  1.  Select kernel
  1. Azure ML compute instance
  2. [compute instance name]

## Specifications

  - VS Code Version: 1.85.0
  - Platform: Linux (EndevourOS), MacOs Ventura 13.6

## Logs
```
Visual Studio Code (1.85.0, undefined, desktop)
Jupyter Extension Version: 2023.11.1003402403.
Python Extension Version: 2023.22.0.
Pylance Extension Version: 2023.11.104.
Platform: linux (x64).

14:08:28.494 [info] Start refreshing Interpreter Kernel Picker
14:08:28.494 [info] Start refreshing Kernel Picker (1702296508494)
14:08:28.538 [warn] No interpreter for Pylance for Notebook URI "*****.ipynb"
14:08:34.446 [info] End refreshing Kernel Picker (1702296508494)
14:09:14.862 [error] Error in validating the Remote Uri azureml_compute_instances./subscriptions/******/resourceGroups/*****/providers/Microsoft.MachineLearningServices/workspaces/******/computes/bko4fe4 [Error: Unable to get resolved server information for ms-toolsai.vscode-ai:azureml_compute_instances:/subscriptions/********/resourceGroups/*******/providers/Microsoft.MachineLearningServices/workspaces/*******/computes/bko4fe4
	at Dn.validateRemoteUri (~/.vscode/extensions/ms-toolsai.jupyter-2023.11.1003402403-linux-x64/dist/extension.node.js:138:8486)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
	at async Jo.addJupyterServer (~/.vscode/extensions/ms-toolsai.jupyter-2023.11.1003402403-linux-x64/dist/extension.node.js:138:10211)
	at async ~/.vscode/extensions/ms-toolsai.jupyter-2023.11.1003402403-linux-x64/dist/extension.node.js:265:125188]
14:09:14.868 [info] Start refreshing Kernel Picker (1702296554868)
14:09:14.914 [error] UniversalRemoteKernelFinder: Failed to get kernels without cache [Error: Unable to get resolved server information for ms-toolsai.vscode-ai:azureml_compute_instances:/subscriptions/**********/resourceGroups/*******/providers/Microsoft.MachineLearningServices/workspaces/*********/computes/bko4fe4
	at Dn.createConnectionInfo (~/.vscode/extensions/ms-toolsai.jupyter-2023.11.1003402403-linux-x64/dist/extension.node.js:138:7497)
	at async ~/.vscode/extensions/ms-toolsai.jupyter-2023.11.1003402403-linux-x64/dist/extension.node.js:235:20130]
14:09:14.956 [info] End refreshing Kernel Picker (1702296554868)
...
```

**Comment by kbokh** [comment]
@shsuman 
Azure Ml Extention: v0.40.0
Jupyter Notebook Renderers: 1.0.17
**Python env:**
notebook:                  7.0.6

**Comment by shsuman** [comment]
@kbokh Can you please provide me with the _Azure ML Trace logs_ by following the steps [here](https://github.com/microsoft/vscode-tools-for-ai/wiki/FAQ#how-to-get-trace-logs-for-azure-machine-learning-extension-) ?

**Comment by kbokh** [comment]
Azure ML Trace logs:
```
[Info - 2023-12-13 09:45:29.363] Using ComputeInstanceJupyterUriProvider
[Info - 2023-12-13 09:46:14.570] Default workspace cache initialized successfully.
[Info - 2023-12-13 09:46:14.787] Default workspace cache synchronized successfully.
[Info - 2023-12-13 09:46:14.845] Default workspace cache synchronized successfully.
[Info - 2023-12-13 10:06:03.805] Default workspace cache synchronized successfully.
[Info - 2023-12-13 10:26:03.050] Default workspace cache synchronized successfully.
[Info - 2023-12-13 10:46:05.586] Default workspace cache synchronized successfully.
[Info - 2023-12-13 11:06:03.101] Default workspace cache synchronized successfully.
[Info - 2023-12-13 11:26:03.553] Default workspace cache synchronized successfully.
[Info - 2023-12-13 11:46:03.073] Default workspace cache synchronized successfully.
[Trace - 2023-12-13 12:00:10.448] AuthenticationService.getAzureMLService called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:00:10.448] AuthenticationService.getSubscriptionById called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:00:10.448] AuthenticationService.listSubscriptions called with arguments: []
[Trace - 2023-12-13 12:00:10.449] Telemetry Event: authentication/needsLogin Measures: undefined Props: {
    "result": "false"
}
[Trace - 2023-12-13 12:01:20.454] AuthenticationService.getAzureMLService called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:01:20.454] AuthenticationService.getSubscriptionById called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:01:20.455] AuthenticationService.listSubscriptions called with arguments: []
[Trace - 2023-12-13 12:01:20.456] Telemetry Event: authentication/needsLogin Measures: undefined Props: {
    "result": "false"
}
[Trace - 2023-12-13 12:02:10.197] ComputeInstanceJupyterUriProvider.getQuickPickEntryItems called with arguments: []
[Trace - 2023-12-13 12:02:15.646] ComputeInstanceJupyterUriProvider.handleQuickPick called with arguments: [
    {
        "label": "Azure ML compute instances",
        "description": "Power your Notebook with an Azure ML compute instance",
        "detail": "Power your Notebook with an Azure ML compute instance",
        "default": true
    },
    true
]
[Trace - 2023-12-13 12:02:15.654] Telemetry Event: remote/computeInstance/jupyterUriProvider/handleQuickPick Measures: undefined Props: {}
[Trace - 2023-12-13 12:02:15.656] AuthenticationService.promptLoginIfNecessaryAndBlock called with arguments: []
[Trace - 2023-12-13 12:02:15.657] Telemetry Event: authentication/needsLogin Measures: undefined Props: {
    "result": "false"
}
[Trace - 2023-12-13 12:02:15.657] AuthenticationService.listSubscriptions called with arguments: [
    null
]
[Trace - 2023-12-13 12:02:15.657] Telemetry Event: authentication/needsLogin Measures: undefined Props: {
    "result": "false"
}
[Trace - 2023-12-13 12:02:16.637] Subscription.listWorkspaces called with arguments: [
    null
]
[Trace - 2023-12-13 12:02:18.248] AuthenticationService.getUserObjectId called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:02:18.248] AuthenticationService.getAuthTokenResponse called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:02:18.248] AuthenticationService.getCredentials called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:02:18.249] AuthenticationService.promptLoginIfNecessaryAndBlock called with arguments: []
[Trace - 2023-12-13 12:02:18.250] Telemetry Event: authentication/needsLogin Measures: undefined Props: {
    "result": "false"
}
[Trace - 2023-12-13 12:02:18.251] AuthenticationService.getSubscriptionById called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:02:18.251] AuthenticationService.listSubscriptions called with arguments: []
[Trace - 2023-12-13 12:02:18.253] Telemetry Event: authentication/needsLogin Measures: undefined Props: {
    "result": "false"
}
[Trace - 2023-12-13 12:02:20.145] ComputeInstanceJupyterUriProvider.getServerUri called with arguments: [
    "/subscriptions/ae5bb58f-e820-4cdd-9e25-dfe765a6edbe/resourceGroups/intrack-dev-ai/providers/Microsoft.MachineLearningServices/workspaces/IntrackDevAIv2/computes/bko4fe4"
]
[Trace - 2023-12-13 12:02:20.145] Telemetry Event: remote/computeInstance/jupyterUriProvider/getServerUri Measures: undefined Props: {}
[Trace - 2023-12-13 12:02:20.146] AuthenticationService.getAzureMLService called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:02:20.146] AuthenticationService.getSubscriptionById called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:02:20.146] AuthenticationService.listSubscriptions called with arguments: []
[Trace - 2023-12-13 12:02:20.149] Telemetry Event: authentication/needsLogin Measures: undefined Props: {
    "result": "false"
}
[Debug - 2023-12-13 12:02:21.314] Ensuring that compute instance bko4fe4 is running
[Debug - 2023-12-13 12:02:21.314] Compute instance bko4fe4 is already running
[Trace - 2023-12-13 12:02:21.317] AuthenticationService.getAuthTokenResponse called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:02:21.317] AuthenticationService.getCredentials called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:02:21.318] AuthenticationService.promptLoginIfNecessaryAndBlock called with arguments: []
[Trace - 2023-12-13 12:02:21.320] Telemetry Event: authentication/needsLogin Measures: undefined Props: {
    "result": "false"
}
[Trace - 2023-12-13 12:02:21.321] AuthenticationService.getSubscriptionById called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:02:21.321] AuthenticationService.listSubscriptions called with arguments: []
[Trace - 2023-12-13 12:02:21.323] Telemetry Event: authentication/needsLogin Measures: undefined Props: {
    "result": "false"
}
[Trace - 2023-12-13 12:02:30.457] AuthenticationService.getAzureMLService called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:02:30.457] AuthenticationService.getSubscriptionById called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:02:30.458] AuthenticationService.listSubscriptions called with arguments: []
[Trace - 2023-12-13 12:02:30.459] Telemetry Event: authentication/needsLogin Measures: undefined Props: {
    "result": "false"
}
[Trace - 2023-12-13 12:03:40.462] AuthenticationService.getAzureMLService called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:03:40.463] AuthenticationService.getSubscriptionById called with arguments: [
    "ae5bb58f-e820-4cdd-9e25-dfe765a6edbe"
]
[Trace - 2023-12-13 12:03:40.463] AuthenticationService.listSubscriptions called with arguments: []
[Trace - 2023-12-13 12:03:40.465] Telemetry Event: authentication/needsLogin Measures: undefined Props: {
    "result": "false"
}

```

**Comment by shsuman** [comment]
Our logs don't show anything weird. Can you please open an issue on https://github.com/microsoft/vscode-jupyter/issues and tag me there. I will follow up with them if need be.

**Comment by shsuman** [comment]
@kbokh Are you unblocked for now ?

**Comment by shsuman** [comment]
Closing in favor of https://github.com/microsoft/vscode-jupyter/issues/14925
